Sources/TuistCore/ValueGraph/ValueGraphTraverser.swift

```diff
@@ -236,7 +236,7 @@ public class ValueGraphTraverser: GraphTraversing {
             if let hostApp = hostApplication(path: path, name: name) {
                 references.subtract(embeddableFrameworks(path: hostApp.path, name: hostApp.target.name))
             } else {
-                references = Set()
+                references = Set(precompiledFrameworks)
```

Comment:
I had to flip the sample project to use an xcframework instead of a framework because of the existing Carthage issues but it worked fine for me. I am actually wondering if your use of lowercase "ios" instead of Carthage using "Build/iOS" is suspect here.

Unless the precompiled frameworks are an actual dependency of the test host itself, I'm looking at a number of our projects and we haven't had this issue and the runpath settings do pick all these up correctly. The behavior I am seeing is that if there is a test host, frameworks get embedded into the test bundle itself, which is correct. The test host should only have frameworks linked/embedded if they are used by code within the test host target.

I think we might want to set up some fixtures around this to make it easier: I did try the test project you linked but it didn't seem like it had a test host,

Reply:
> I had to flip the sample project to use an xcframework instead of a framework because of the existing Carthage issues but it worked fine for me.

Command `tuist up` should work fine as it uses `carthage.sh` scrirpt instead of Carthage directly. I agree that using `xcframework` is the right way to go but this requires all dependencies migrated to `xcframework`.

> I am actually wondering if your use of lowercase "ios" instead of Carthage using "Build/iOS" is suspect here.

Hmm, looks like it's actually the root cause of the issue. Great catch. Thanks a lot!